src/plugin.js

```
@@ -7,16 +7,25 @@
  * of a Microcosm.
  */
 
-const defaults = {
-  __start(callback) {
-    this.register(this.app, this.options, callback)
-  },
+function PluginFactory(config, options, app) {
+  return Object.assign({ app, options }, config)
+}
+
+function installPlugin (next, plugin) {
+  return function (error) {
+    // Halt execution of all future plugin installation if there is an error
+    if (error) {
+      return next(error)
+    }
 
-  register(app, options, next) {
-    next(null)
+    // Plugins might not have a register method. In this case, just continue through
+    return plugin.register ? plugin.register(plugin.app, plugin.options, next) : next(null)
```

Hmmm. That would kind of be a silent error though. What about a dev-only error?